zerver/tests/email_change_test_helper.py

```diff
@@ -0,0 +1,206 @@
+import datetime
+from typing import Optional
+
+from django.utils.timezone import now as timezone_now
+
+from confirmation.models import Confirmation, generate_key, get_object_from_key
+from zerver.lib.actions import do_set_realm_property
+from zerver.lib.create_user import create_user_profile
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.models import (
+    EmailChangeStatus,
+    Realm,
+    UserProfile,
+    get_realm,
+    get_user_by_delivery_email,
+)
+
+
+def create_email_change_confirmation_key(user_profile: UserProfile, new_email: str,
+                                         date_sent: Optional[datetime.datetime]=None) -> str:
+    old_email = user_profile.delivery_email
+    new_email = new_email
+    obj = EmailChangeStatus.objects.create(new_email=new_email,
+                                           old_email=old_email,
+                                           user_profile=user_profile,
+                                           realm=user_profile.realm)
+    key = generate_key()
+    if date_sent is None:
+        date_sent = timezone_now()
+    Confirmation.objects.create(content_object=obj,
+                                date_sent=date_sent,
+                                confirmation_key=key,
+                                type=Confirmation.EMAIL_CHANGE)
+    return key
+
+class EmailChangeTestMixin(ZulipTestCase):
```

Comment:
ohh, this is really cool, kinda like the social auth flow test base class. I didn't expect this would generalize into a base class so nicely. Though  I think this should go to `test_classes.py`, we have some other general "library" classes there.

Reply:
sure, glad you liked it, moving it now. Does the name sound good? or do we want to go for something like `EmailChangeTestBase` or `EmailChangeBase`?

Reply:
Moved it to `test_classes.py`.